Add OS-Agent to system requirements of Supervised (#584)
home-assistant/architecture: adr/0014-home-assistant-supervised.md

- NetworkManager >= 1.14.6
- AppArmor == 2.13.x (built into the kernel)
- Debian Linux Debian 10 aka Buster (no derivatives)

> - Home Assistant OS-Agent (close as possible to latest release)

Reviewer: ```suggestion
- [Home Assistant OS-Agent](https://github.com/home-assistant/os-agent) (Only the [latest release](https://github.com/home-assistant/os-agent/releases/latest) is supported)
```

Author: Right, we don't need force so strict the requirements but at least it's still clear what is need to do

Author: I mean, the suggestion is fine for me :)
